From ebiederm@xmission.com  Wed Jul 21 13:33:47 2010
To: Greg KH <gregkh@suse.de>
Cc: Andrew Morton <akpm@linux-foundation.org>, Greg KH <greg@kroah.com>,
        "Rafael J. Wysocki" <rjw@sisk.pl>,
        "Maciej W. Rozycki" <macro@linux-mips.org>,
        Kay Sievers <kay.sievers@vrfy.org>,
        Johannes Berg <johannes@sipsolutions.net>
Subject: sysfs: allow creating symlinks from untagged to tagged directories
From: ebiederm@xmission.com (Eric W. Biederman)
Date: Tue, 20 Jul 2010 22:12:01 -0700
Message-ID: <m1r5ix4dwe.fsf_-_@fess.ebiederm.org>


Supporting symlinks from untagged to tagged directories is reasonable,
and needed to support CONFIG_SYSFS_DEPRECATED.  So don't fail a prior
allowing that case to work.

Signed-off-by: Eric W. Biederman <ebiederm@xmission.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/symlink.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/sysfs/symlink.c
+++ b/fs/sysfs/symlink.c
@@ -67,7 +67,8 @@ static int sysfs_do_create_link(struct k
 
 	sysfs_addrm_start(&acxt, parent_sd);
 	/* Symlinks must be between directories with the same ns_type */
-	if (ns_type == sysfs_ns_type(sd->s_symlink.target_sd->s_parent)) {
+	if (!ns_type ||
+	    (ns_type == sysfs_ns_type(sd->s_symlink.target_sd->s_parent))) {
 		if (warn)
 			error = sysfs_add_one(&acxt, sd);
 		else
